src/core/Modal/Modal/Modal.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { Component, ReactNode, MouseEvent } from 'react';
+import ReactDOM from 'react-dom';
+import { default as styled } from 'styled-components';
+import { createFocusTrap, FocusTrap } from 'focus-trap';
+import classnames from 'classnames';
+import { HtmlDiv, HtmlDivProps } from '../../../reset';
+import { AutoId } from '../../../utils/AutoId';
+import { windowAvailable } from '../../../utils/common';
+import { ModalContent, ModalFooter } from '../';
+import './Modal.globals.scss';
+import { baseStyles } from './Modal.baseStyles';
+
+export type ModalVariant = 'smallScreen' | 'default';
+
+export interface ModalProps
+  extends Omit<HtmlDivProps, 'children' | 'className' | 'title' | 'ref'> {
+  /** Show or hide the modal */
+  visible: boolean;
+  /** Modal container div class name for custom styling. */
+  className?: string;
+  /** Children */
+  children: ModalContent | ModalFooter | ReactNode;
+  /**
+   * Variant. Use smallScreen for mobile and small displays
+   * @default 'default'
+   */
+  variant?: ModalVariant;
+  /**
+   * Show vertical scroll bar if needed and show horizontal divider before buttons.
```

Comment:
Might be clearer to replace 'before buttons' with 'between content and footer' since technically footer doesn't require a button.

Reply:
Modal requires at least one focusable element, but it does not have to be in the footer. If there are no focusable elements, focus trap will throw an error.

Reply:
Fixed according to suggestion.